### PR TITLE
Respect local time setting when exporting balances snapshots

### PR DESCRIPTION
**Pull request by nebolax**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/tests/api/test_snapshots.py [hunk 1]
```diff
@@ -279,12 +281,37 @@ def _create_snapshot_with_invalid_headers(directory: str, timestamp: Timestamp)
         _write_location_data_csv_row_with_invalid_headers(writer, timestamp)
 
 
+def validate_timestamp(
+        received_serialized_timestamp: str,
+        expected_utc_timestamp: Timestamp,
+        display_date_in_localtime: bool,
+        is_for_import: bool,
+):
+    if is_for_import:  # Should always be a numeric utc timestamp
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_snapshots.py:290]: this is missing a docstring about what it does

File: rotkehlchen/tests/api/test_snapshots.py [hunk 2]
```diff
@@ -279,12 +281,37 @@ def _create_snapshot_with_invalid_headers(directory: str, timestamp: Timestamp)
...
+        if display_date_in_localtime:  # Check that respects the timezone
+            expected_datetime = datetime.datetime.fromtimestamp(
+                expected_utc_timestamp,
+                tz=datetime.datetime.now(datetime.timezone.utc).astimezone().tzinfo,
+            ).strftime('%Y-%m-%d %H:%M:%S')
+        else:  # Should be a stringified utc timestamp
+            expected_datetime = datetime.datetime.fromtimestamp(
+                expected_utc_timestamp,
+                tz=datetime.timezone.utc,
+            ).strftime('%Y-%m-%d %H:%M:%S')
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_snapshots.py:303]: I believe this two lines in the CI are the same

**Review by yabirgb — COMMENTED** [review]
two comments about the test

File: rotkehlchen/tests/api/test_snapshots.py [hunk 3]
```diff
@@ -279,12 +282,43 @@ def _create_snapshot_with_invalid_headers(directory: str, timestamp: Timestamp)
...
+
+    If is_for_import is True, then the timestamp should be a utc timestamp in seconds.
+    If display_date_in_localtime is True, then timestamp should be stringified in local timezone.
+    Otherwise the timestamp should be stringified in utc.
+    """
+    if is_for_import:  # Should always be a numeric utc timestamp
+        deserialized_timestamp = deserialize_timestamp(received_serialized_timestamp)
+        assert deserialized_timestamp == expected_utc_timestamp
+    else:
+        if display_date_in_localtime:  # Check that respects the timezone
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_snapshots.py:301]:
```suggestion
        if display_date_in_localtime:  # Check that it respects the timezone
```

**Review by LefterisJP — APPROVED** [review]
LGTM with a tiny wording nitpick.

Just not sure on frontend tests. Waiting for feedback from @tewshi

File: rotkehlchen/tests/api/test_snapshots.py (hunk 3 above)
Comment (tewshi) [reply rotkehlchen/tests/api/test_snapshots.py:301]: it is all good, you can skip and proceed. @LefterisJP 

@nebolax when you fix the above suggestion, push again, and let's see.

**Pull request merged**